src/main/java/hudson/plugins/emailext/RecipientListStringAnalyser.java

```diff
@@ -0,0 +1,101 @@
+package hudson.plugins.emailext;
+
+import javax.mail.internet.InternetAddress;
+import java.text.MessageFormat;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class RecipientListStringAnalyser {
+
+    private static final Logger LOG = Logger.getLogger(RecipientListStringAnalyser.class.getName());
```

Comment:
I know its not so nice but there is a debug logger thingy used else ware that enables switching on debugging in settings so debug logs likes these end up in the build output. If its not too hard would prefer to use that

Reply:
(Sorry for the delay)
After some looking/searching I guess you refer to the "Configure System" setting "Enable Debug Mode" for "Extended E-mail plugin"? => I need to use `ExtendedEmailPublisherContext` and `ExtendedEmailPublisherDescriptor`?

Reply:
Well, after much more looking I dare to say that there is one code dependency that I think does not allow logging via that approach:
* Dependencies:
```
RecipientListStringAnalyser
  static hudson.plugins.emailext.EmailRecipientUtils#convertRecipientString(java.lang.String, hudson.EnvVars, int)
    static hudson.plugins.emailext.EmailRecipientUtils#convertRecipientString(java.lang.String, hudson.EnvVars)
    hudson.plugins.emailext.EmailRecipientUtils#validateFormRecipientList
      hudson.plugins.emailext.ExtendedEmailPublisherDescriptor#doRecipientListRecipientsCheck
    static hudson.plugins.emailext.EmailRecipientUtils#addAddressesFromRecipientList
```
  * My new `RecipientListStringAnalyser` class is only used by `EmailRecipientUtils` class
  * But this in turn is used mostly statically, but not always; and also does not always have an `ExtendedEmailPublisherContext` instance or its task listener (which in turn holds the logger): namely the problem is in the `hudson.plugins.emailext.EmailRecipientUtils#validateFormRecipientList` dependency

The resulting code will not be very nice, because the logging must be conditionally on the availability of the logger and so no logging for `hudson.plugins.emailext.EmailRecipientUtils#validateFormRecipientList`/`hudson.plugins.emailext.ExtendedEmailPublisherDescriptor#doRecipientListRecipientsCheck` is possible, I think.

Since I don't know what this is used for I am not sure if I should try that (non-perfect/-elegant approach via the conditionally logging)?

Reply:
Oh, and I would need to break compatibility of `hudson.plugins.emailext.EmailRecipientUtils`, I think; but maybe that is OK anyhow?